dateparser/parser.py

```diff
@@ -430,16 +431,11 @@ def _correct_for_day(self, dateobj):
             return dateobj
 
         _, tail = calendar.monthrange(dateobj.year, dateobj.month)
```

Comment:
I have just seen that we don't really need to generate here the `tail` as can be directly generated in the `set_correct_day_from_settings`. I will change it when I have a moment.